Libplanet/Blockchain/BlockChain.cs

```diff
@@ -143,11 +145,14 @@ bool inFork
                     "restarted the chain with a new genesis block so that it is incompatible " +
                     "with your existing chain in the local store.";
                 throw new InvalidGenesisBlockException(
-                    networkExpected: genesisBlock.Hash,
-                    stored: Genesis.Hash,
+                    networkExpected: GenesisHash,
+                    stored: Store.IterateIndexes(Id, limit: 1).First(),
```

Comment:
Is this operation necessary to be calculated twice?  The already calculated value in the `if` condition seems reusable here.